src/Bicep.Core/Semantics/Namespaces/ExtensionNamespaceTypeHelper.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using Bicep.Core.Features;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.TypeSystem.Types;
+
+namespace Bicep.Core.Semantics.Namespaces
+{
+    public static class ExtensionNamespaceTypeHelper
+    {
+        public static ImmutableArray<NamedTypeProperty> GetExtensionNamespaceObjectProperties(NamespaceSettings namespaceSettings, IFeatureProvider features)
+        {
+            if (features is not { ExtensibilityEnabled: true, ModuleExtensionConfigsEnabled: true } || namespaceSettings.ConfigurationType is null)
+            {
+                return [];
+            }
+
+            var namespaceProperties = ImmutableArray.CreateBuilder<NamedTypeProperty>();
+
+            var namespaceConfigType = TypeHelper.DeepCloneAndModifyPropertyFlagsRecursively(namespaceSettings.ConfigurationType, f => f | TypePropertyFlags.ReadOnly);
```

Comment:
Do you need this at all? Bicep doesn't allow mutations, so I can't think why you'd need to explicitly mark properties as readonly.